java/client/src/main/java/glide/managers/ConnectionManager.java

```diff
@@ -141,6 +144,19 @@ private ConnectionRequest.Builder setupConnectionRequestBuilderRedisClient(
             connectionRequestBuilder.setDatabaseId(configuration.getDatabaseId());
         }
 
+        if (configuration.getSubscriptionConfiguration() != null) {
+            var subscriptionsBuilder = PubSubSubscriptions.newBuilder();
```

Comment:
I would move this logic to the client, and create & the subscriptions in the constructor.  Then you can just use a getter here to grab the subscriptions.  

Would reduce duplicate code below.

Reply:
`PubSubSubscriptions` is a protobuf class. It is a right place for that.